clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/config/BootstrapKubernetesConfigurationProvider.java

```diff
@@ -0,0 +1,148 @@
+import com.netflix.spinnaker.kork.configserver.CloudConfigResourceService;
+import com.netflix.spinnaker.kork.configserver.ConfigFileLoadingException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.beans.BeansException;
+import org.springframework.cloud.bootstrap.config.BootstrapPropertySource;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * For larger number of Kubernetes accounts, SpringBoot implementation of properties binding is
+ * inefficient, hence a custom constructor for KubernetesConfigurationProperties is written. This
+ * class fetches the flattened kubernetes properties from Spring Cloud Config's
+ * BootstrapPropertySource and creates a KubernetesConfigurationProperties object. Also adds support
+ * to existing "configserver:" feature along with caching ability. The main objective of this
+ * implementation is to reduce the clouddriver loading time when large number of dynamic accounts
+ * are configured. So some of the features provided by SpringBoot and clouddriver modules do not
+ * work. Here are the limitations: 1. The properties in the yaml file must use camel case 2.
+ * Property placeholders don't work. 3. "encryptedFile:" and "encrypted:" notations don't work 4.
+ * "configserver:" works only for kubeconfig files
+ */
+public class BootstrapKubernetesConfigurationProvider implements KubernetesConfigurationProvider {
+
+  private final ConfigurableApplicationContext applicationContext;
+  private CloudConfigResourceService configResourceService;
+  private Map<String, String> configServerCache;
+
+  public BootstrapKubernetesConfigurationProvider(
+      ConfigurableApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
+  }
+
+  @Override
+  public KubernetesConfigurationProperties getKubernetesConfigurationProperties() {
+    return getKubernetesConfigurationProperties(getPropertiesMap());
+  }
+
+  @SuppressWarnings("unchecked")
+  public KubernetesConfigurationProperties getKubernetesConfigurationProperties(
+      Map<String, Object> kubernetesPropertiesMap) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    // remove the keys having blank string values
+    kubernetesPropertiesMap.values().removeAll(Collections.singleton(""));
+
+    Map<String, Object> propertiesMap =
+        (Map<String, Object>)
+            JsonUnflattener.unflattenAsMap(kubernetesPropertiesMap).get("kubernetes");
+
+    KubernetesConfigurationProperties kubernetesConfigurationProperties =
+        objectMapper.convertValue(propertiesMap, KubernetesConfigurationProperties.class);
+
+    kubernetesConfigurationProperties
+        .getAccounts()
+        .forEach(
+            acc -> {
+              if (acc.getKubeconfigFile().startsWith("configserver:")) {
```

Comment:
I'm not sure I understand why we need to post-process only this field on the account?  Could you please briefly explain (and maybe add a comment for this).

Reply:
I though only kubeconfigFile would need the configserver feature but now I extended the logic to all the fields.